### PR TITLE
introduce node.UseGlobalArtifacts

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:
introduce node.UseGlobalArtifacts

**Special notes for your reviewer**:

**Release note**:

```
Add support for the usage of GlobalArtifacts in steps
```

### CODE REVIEW

File: pkg/apis/testmachinery/v1beta1/types.go [hunk 1]
```diff
@@ -232,10 +232,11 @@ type ConfigElement struct {
 type TestFlow []*DAGStep
 
 type DAGStep struct {
-	Name          string         `json:"name,omitempty"`
-	Definition    StepDefinition `json:"definition,omitempty"`
-	DependsOn     []string       `json:"dependsOn,omitempty"`
-	ArtifactsFrom string         `json:"artifactsFrom,omitempty"`
+	Name               string         `json:"name,omitempty"`
+	Definition         StepDefinition `json:"definition,omitempty"`
+	UseGlobalArtifacts bool           `json:"useGlobalInput,omitempty"`
```
Comment (schrodit) [inline pkg/apis/testmachinery/v1beta1/types.go]: Adjust json name

File: pkg/testmachinery/testflow/node/node.go [hunk 2]
```diff
@@ -169,6 +147,54 @@ func (n *Node) Status() *tmv1beta1.StepStatus {
 	return status
 }
 
+func (n *Node) isRootNode() bool {
+	return n.inputSource != nil
+}
+
+func (n *Node) GlobalArtifacts() []argov1.Artifact {
```
Comment (schrodit) [inline pkg/testmachinery/testflow/node/node.go]: `GlobalArtifacts` is misleading.
We should name something like `generateArtifacts` or `getTaksArtifacts`

File: pkg/testmachinery/testflow/node/node.go [hunk 3]
```diff
@@ -169,6 +147,54 @@ func (n *Node) Status() *tmv1beta1.StepStatus {
...
 }
 
+func (n *Node) isRootNode() bool {
+	return n.inputSource != nil
+}
+
+func (n *Node) DetermineArtifacts() []argov1.Artifact {
+	artifactsMap := make(map[string]argov1.Artifact)
+
+	if n.step.UseGlobalArtifacts {
```
Comment (schrodit) [inline pkg/testmachinery/testflow/node/node.go]: if we use maps then we could move the global artifact down and remove the if in line 168.
Then we can get rid of one if condition and have a more semantic code that global artifacts are not the default.
```go
if n.isRootNode() {
  artifactMap["kubeconfig"]...
  ...
}
if n.step.UseGlobalArtifacts {
  ...
}
```

File: pkg/testmachinery/testflow/node/node.go (hunk 2 above)
Comment (OlegLoewen) [reply pkg/testmachinery/testflow/node/node.go]: done

File: pkg/testmachinery/testflow/node/node.go (hunk 3 above)
Comment (OlegLoewen) [reply pkg/testmachinery/testflow/node/node.go]: done

**Pull request merged**